target_chains/ethereum/sdk/js/src/tracer/debug-trace-call.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Address,
+  BlockTag,
+  CallParameters,
+  Client,
+  decodeFunctionData,
+  ExactPartial,
+  formatTransactionRequest,
+  Hex,
+  RpcTransactionRequest,
+} from "viem";
+
+import { IPythAbi } from "../pyth-abi";
+
+/**
+ * Extract Pyth price feed IDs from a transaction call trace.
+ */
+export function extractPythPriceFeedsFromDebugTraceCall(
+  trace: RpcCallTrace,
+  pythContractAddress: Address,
+  ignoreParsingErrors = false,
+): Set<`0x${string}`> {
+  const result = new Set<`0x${string}`>();
+  if (trace.to === pythContractAddress) {
+    // Decode the calldata to see what function is being called
+    try {
+      const decoded = decodeFunctionData({
+        abi: IPythAbi,
+        data: trace.input,
+      });
+
+      let priceFeedId: `0x${string}` | undefined;
+      switch (decoded.functionName) {
+        case "getPrice":
+        case "getPriceNoOlderThan":
+        case "getPriceUnsafe":
+        case "getEmaPrice":
+        case "getEmaPriceNoOlderThan":
+        case "getEmaPriceUnsafe": {
+          priceFeedId = decoded.args[0];
+          break;
+        }
+        default: {
+          break;
+        }
+      }
+      if (priceFeedId !== undefined) {
+        result.add(priceFeedId);
+      }
+    } catch {
+      if (!ignoreParsingErrors) {
+        throw new Error(
+          `Failed to decode calldata: ${trace.input}. Make sure correct Pyth contract address is used.`,
+        );
+      }
+    }
```

Comment:
```suggestion
    } catch (error: unknown) {
      if (!ignoreParsingErrors) {
        const thrownError = new Error(
          `Failed to decode calldata: ${trace.input}. Make sure correct Pyth contract address is used.`,
        );
        thrownError.cause = error;
        throw thrownError;
      }
    }
```

this way you don't lose the context of the underlying error